### PR TITLE
Product Listing page

### PR DESCRIPTION
**Pull request by anshikaa12**
> fetched the data from the api and displayed it on product listing page
>not responsive yet
ecomm_productlistReactt

### CODE REVIEW

File: src/components/Card.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+function Card() {
+
+    const [cardData, setCardData] = useState([]);
+    useEffect(() => {
+        axios.get('/api/products').then((data) => {
+            console.log("YO");
+            console.log(data.data.products);
```
Comment (sharathnair9999) [inline src/components/Card.jsx]: remove console logs after testing

File: src/components/Nav.jsx [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 
 function Nav() {
     return (
         <div>
+
             <div class="nav">
                 <div class="first-nav">
                     <img src="https://res.cloudinary.com/de5it79dw/image/upload/v1648491301/Pixelion_2_-PhotoRoom_q7mfmo.png" alt="" />
                 </div>
                 <div class="mid-nav flex-row">
```
Comment (sharathnair9999) [inline src/components/Nav.jsx]: use className instead of class

**Review by sharathnair9999 — APPROVED** [review]
Suggested minor changes

**Review by abhijeetekad — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
 import "./App.css";
 import Nav from "./components/Nav";
 import Home from "./pages/home/Home";
-
+import { Route, Routes } from "react-router-dom";
+import Product from "./pages/product_listing/product";
```
Comment (arp99) [inline src/App.js]: Suggestion: You can put a `index.js` file inside pages folder and export all the pages from there.
Then you don't need to write this long import line for each page:
Currently for Home and Product, you are doing:
```javascript
import Home from "./pages/home/home"
import Product from "./pages/product_listing/product"
```
After doing the suggestions you just have to do:
```javascript
import {  Home, Product } from "./pages"
```

File: src/components/Card.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+function Card() {
+    const [cardData, setCardData] = useState([]);
+    useEffect(() => {
+        axios.get('/api/products').then((data) => {
+            setCardData(data.data.products);
+        }).catch((err) => { console.error(err) });
```
Comment (arp99) [inline src/components/Card.jsx]: Suggestion:
For such api calls make a folder names `services` and make utility functions for these and import wherever you need

File: src/components/Card.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }, []);
+
+    return (
+
+        cardData && cardData.map(item => {
+            return (
+                <div className="e-basic-card">
+                    <img src={item.image} className="e-card-img" />
```
Comment (arp99) [inline src/components/Card.jsx]: Add a alt attribute in the image

File: src/components/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                            <div className="connect-part"><i className="far fa-heart card-icon"></i></div>
+                        </div>
+                    </div>
+                </div>
+            )
+        })
+
+
+
+    )
```
Comment (arp99) [inline src/components/Card.jsx]: Use prettier to make your indentation right and remove extra spaces

File: src/pages/product_listing/product.jsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import Card from '../../components/Card'
+import Footer from '../../components/Footer'
+import Nav from '../../components/Nav'
```
Comment (arp99) [inline src/pages/product_listing/product.jsx]: Suggestion: Similarly as said above for pages, you can export each component from `index.js` file inside components

**Review by arp99 — COMMENTED** [review]
Added some suggestions and sanity checks

File: src/App.js [hunk 8]
```diff
@@ -1,8 +1,6 @@
 import "./App.css";
-import Nav from "./components/Nav";
-import Home from "./pages/home/Home";
 import { Route, Routes } from "react-router-dom";
-import Product from "./pages/product_listing/product";
+import {Home,Product} from "./pages/index"
```
Comment (arp99) [inline src/App.js]: It can be directly imported as :
```javascript
import { Home, Product} from "./pages"
```
Try it once

**Review by arp99 — COMMENTED** [review]
Requested one change, rest looks good

**Pull request merged**